samtranslator/model/api/api_generator.py

```diff
@@ -459,6 +469,129 @@ def _add_auth(self):
 
         self.definition_body = self._openapi_postprocess(swagger_editor.swagger)
 
+    def _construct_usage_plan(self, rest_api_stage=None):
+        """Constructs and returns the ApiGateway UsagePlan, ApiGateway UsagePlanKey, ApiGateway ApiKey for Auth.
+
+        :param model.apigateway.ApiGatewayStage stage: the stage of rest api
+        :returns: the UsagePlan, UsagePlanKey, ApiKey for this SAM Api
+        :rtype: model.apigateway.ApiGatewayUsagePlan, model.apigateway.ApiGatewayUsagePlanKey,
+                model.apigateway.ApiGatewayApiKey
+        """
+        if not self.auth:
+            return []
+        auth_properties = AuthProperties(**self.auth)
+        if auth_properties.UsagePlan is None:
+            return []
+        usage_plan_properties = auth_properties.UsagePlan
+
+        # throws error if the property invalid/ unsupported for UsagePlan
+        if not all(key in UsagePlanProperties._fields for key in usage_plan_properties.keys()):
```

Comment:
we don't error on unsupported properties today, so we shouldnt be adding this check here. We only validate that we have the required properties set.

Reply:
We are throwing errors for [invalid Auth properties](https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/api/api_generator.py#L312). So, I have thrown errors for UsagePlan properties as well

Reply:
I like this here 👍